### DISCUSSION TITLE
New dev version

### DISCUSSION THREAD

**Discussion by susuhahnml** [potassco/clingraph#56] [Announcements]
# New dev version :sun_with_face:

The new development version reorganized the library structure and added new features in the command line. These changes can be fond bellow or in the [documentation](https://clingraph.readthedocs.io/en/latest/ ). Check the [examples folder](https://github.com/potassco/clingraph/tree/master/examples) now with the image outputs for each example to see clingraphs' potential.

### Changes in the command line

- **New single output argument `--out`**: This removes flags: `--render` `--tex` `--gif`.
   - `out=facts`: Prints preprocessed facts
   - `out=dot`: Prints string in DOT language
   - `out=render`: Generates images with the rendering method of graphviz
   - `out=gif`: Generates a gif after rendering
   - `out=tex`: Prints a latex file
- **Deterministic graphs**: The generated graphs will now be more stable on the way they are printed from one run to another. Additionally the new `--seed` argument allows to set the seed used in clingo and in the engine for rendering.
- **Save output**: The facts, dot and tex are now outputted to stdout and can be saved in files using argument `--save`
- **New name format for output**: The new `--name-format` argument sets the format of the name of the output file and can include `{graph_name}` and `{model_name}`. This removes the older `--name-prefix` argument.
- **Model selection**: The model selection now starts on index `0`
- **Gifs for multi model**: Now gifs can be generated for multiple models
- **Gifs params**: The argument `--gif-param` is removed and the argument `--fps` was added
- **Sorting gif images**. The images of a gif can now be sorted with the `--sort` argument
- **Visualization encoding**: The visualization encoding can now also be passed as an argument `--viz-encoding` and will be applied to multiple models if a json is piped.
- **Removed --json**: The `--json` parameter is not necesary anyome, it is now detected automatically.

### Changes in the python package

- **Clingraph is now Factbase**: The Clingraph class was replaced by the new class is Factbase with the same functionality to add the facts and strings. This class doesn't have the `compute_graphs` and `save` method. This are now part of the graphviz sub-package.
- **Graphviz sub-package**: The graphviz sub-package ``clingraph.graphviz`` now has the standalone functions `compute_graphs`, `dot` and `render`
- **Graphviz sub-package tex**: The graphviz sub-package ``clingraph.graphviz.tex`` now has the standalone functions `tex` to generate latex code
- **Graphviz sub-package imageio**: The graphviz sub-package ``imageio`` now has the standalone functions `save_gif` to generate gifs
- **MultiModelClingraph removed**: This class was removed and now functions accept a list as input for multi models or a dictionary for single models.

### Bugs

- **Escaped characters**: Now escaped characters are parsed differently. The json integration for escaped characters is pending fixes of clasp json output. 
- **Graphviz version**: The minimal graphviz version was set in the requirements to avoid issues when rendering

**Comment by sthiele** [comment]
Thank you @susuhahnml! This looks pretty cool!

**Comment by susuhahnml** [comment]
There were some errors in the installation. It should now be fixed in case you want to try it again 👍

**Comment by susuhahnml** [comment]
Yet another version of clingraph is out. Thanks to the feedback of @MaxOstrowski  and @florianfisher91.
- Accepts jsons with results other than "SATISFIABLE"
- The paths where the files are saved are now returend by the package functions and only printed when using command line.
